### ISSUE TITLE
Add typecheck (int, float, str) to aocd.models.puzzle.answer_a/b?

### ISSUE THREAD

**Issue by Ryton** [wimglenn/advent-of-code-data#97]
An answer was rejected when send through aocd, when using a **float** (w/o siginificant digits behind comma) as an input,  while it was accepted afterwards from the website, with an **int** as an input, with the same value.  AFAIK, AoC never had any non-whole numbers as the answer in the past?
Expected behaviour: 
An int is sent to the website, rather than the float

Potential solution:
Idk how to best catch this unexpected behaviour.
Should the library always send an int?  Or do input typecheck on top of valuecheck and give a warning?Maybe change a float answer with mod(X,1)==0 (no significant digits behind comma)  to INT? 

Example for 6/'22, part A:
![image](https://user-images.githubusercontent.com/6223082/206266259-f7987463-71f4-4c24-a5c5-f67357c1139d.png)
![image](https://user-images.githubusercontent.com/6223082/206266177-0bdd38fd-5135-46a6-b5fb-b2d6a3894669.png)

Same behaviour for other days 15/'21, for example part B: 
![image](https://user-images.githubusercontent.com/6223082/206268662-d58c7fa0-8adc-4765-b9df-07755669e694.png)
![image](https://user-images.githubusercontent.com/6223082/206268586-5db65729-5d54-44e9-9ac7-e17e36192051.png) was accepted
![image](https://user-images.githubusercontent.com/6223082/206269015-e703b962-d72a-4619-9267-51147dc71cda.png)

if you need more info, let me know.
(& kudos for the amazing library!)

**Comment by wimglenn** [comment]
Hey @Ryton, that's unfortunate, I think we could safely coerce floats which exactly represent integers in aocd. It would be adding something like this [here](https://github.com/wimglenn/advent-of-code-data/blob/aa0688d7a09f6a5a1817d29a161df99ddc0b0122/aocd/models.py#L328):

```
if isinstance(val, float) and val.is_integer():
    log.warning("coercing float value")
    val = int(val)
```

What do you think?

**Comment by wimglenn** [comment]
I think this is a reasonable compromise- the warning should indicate to a user that they probably want to be careful with their answer type, without the possibility of them getting the dreaded "That's not the right answer" response on a technicality.

I will add it when I find the time - or if you would like to prepare a PR to get it in right away, be my guest.

**Comment by Ryton** [comment]
I could give it a try, but it will need to be thoroughly checked by you, as i'm not as fluent in git-PR'ing.

So yes, challenge accepted! :-)

**Comment by Ryton** [comment]
TIL how to use the PR cycle ;-) 
Thank you! Both AOCD and AoC great learning aids! :-)

**Comment by wimglenn** [comment]
If you write "closes #97" somewhere in your PR description (you can edit it in), then this issue will be closed automatically by github when the PR is merged.
